egs2/TEMPLATE/tts2/tts2.sh

```diff
@@ -0,0 +1,1283 @@
+ngpu=1               # The number of gpus ("0" uses cpu, otherwise use gpu).
+num_nodes=1          # The number of nodes.
+nj=32                # The number of parallel jobs.
+inference_nj=32      # The number of parallel jobs in decoding.
+gpu_inference=false  # Whether to perform gpu decoding.
+dumpdir=dump         # Directory to dump features.
+expdir=exp           # Directory to save experiments.
+python=python3       # Specify python to execute espnet commands.
+
+# Data preparation related
+local_data_opts="" # Options to be passed to local/data.sh.
+
+# Feature extraction related
+feats_type=raw             # Input feature type.
+audio_format=flac          # Audio format: wav, flac, wav.ark, flac.ark  (only in feats_type=raw).
+min_wav_duration=0.1       # Minimum duration in second.
+max_wav_duration=20        # Maximum duration in second.
+use_sid=false              # Whether to use speaker id as the inputs (Need utt2spk in data directory).
+use_lid=false              # Whether to use language id as the inputs (Need utt2lang in data directory).
+feats_extract=fbank        # On-the-fly feature extractor.
+feats_normalize=global_mvn # On-the-fly feature normalizer.
+fs=16000                   # Sampling rate.
+n_fft=1024                 # The number of fft points.
+n_shift=256                # The number of shift points.
+win_length=null            # Window length.
+fmin=80                    # Minimum frequency of Mel basis.
+fmax=7600                  # Maximum frequency of Mel basis.
+n_mels=80                  # The number of mel basis.
+# Only used for the model using pitch & energy features (e.g. FastSpeech2)
+f0min=80  # Maximum f0 for pitch extraction.
+f0max=400 # Minimum f0 for pitch extraction.
+
+# Speech Discretization Related
+discrete_stage=1                # discretization stage
+discrete_stop_stage=3           # discretization stop stage
+clustering_nj=20                # Number of threads used for kmeans clustering
+feature_dir="dump/feats"        # Feature directory for dumped feature
+km_tag=                         # KMeans tagging
+use_gpu_feat_extract=true       # Whether to use gpu for feature extraction
+feature_layer=6                 # Layers for feature extraction
+s3prl_upstream_name=hubert      # S3PRL upstream name for feature extraction
+feature_clustering_tool="sklearn" # Tool for feature clustering (sklearn or faiss or cuml)
+clustering_portion=0.1          # the portion of data used for clustering
+feature_num_clusters=500        # Number of feature clusters
+storage_save_mode=true          # Save storage on SSL feature extraction
+                                # If true, feature extraction and kmeans clustering on the fly
+
+
+# X-Vector related
+use_xvector=false   # Whether to use x-vector.
+xvector_tool=kaldi  # Toolkit for extracting x-vector (speechbrain, rawnet, espnet, kaldi)
+xvector_model=speechbrain/spkrec-ecapa-voxceleb  # For only espnet, speechbrain, or rawnet
+
+# Vocabulary related
```

Comment:
Are oov, blank, sos_eos necessary?

Reply:
Yeah, this is used in the text tokenizer. Though blank is not necessary for TTS, we still use it to align with ASR/TTS side of the tokenizer.